File: client/src/app/models/predmet.ts
import { Student } from "./student";

export interface Predmet {
    id: number;
    zkratka: string;
    katedra: string;
    nazev: string;
    kreditu?: number;
    rok: string;
    statut: string;
    doporucenyRocnik?: number;
    doporucenySemestr: string;
    vyznamPredmetu: string;
    vyukaLS: string;
    vyukaZS: string;
    rozsah: string;
    typZk: string;
    oborIdNum?: number;
    files: Soubor[];
  }

  export interface Soubor {
    id: number;
    url: string;
    fileName: string;
    extension: string;
    dateAdded: string;
    studentName: string;
    accountName: string;
    studentsLikedBy: SouborLike[];
  }

  export interface Hodnoceni {
    id: number;
    text: string;
    rating: number;
    accountName: string;
    created: string;
    edited: string;
    predmet: Predmet;
    studentsLikedBy: HodnoceniLike[];
  }

  export interface SouborLike {
    souborId: number;
    studentId: number;
  }

  export interface HodnoceniLike {
    hodnoceniId: number;
    studentId: number;
  }

File: API/Entities/Predmet.cs
using System.Collections.Generic;

namespace API.Entities
{
    public class Predmet    {
        public Predmet() => this.Files = new List<Soubor>();
        public int ID { get; set; }
        public string zkratka { get; set; } 
        public string katedra { get; set; } 
        public string nazev { get; set; } 
        public int? kreditu { get; set; } 
        public string rok { get; set; } 
        public string statut { get; set; } 
        public int? doporucenyRocnik { get; set; } 
        public string doporucenySemestr { get; set; } 
        public string vyznamPredmetu { get; set; } 
        public string vyukaLS { get; set; } 
        public string vyukaZS { get; set; } 
        public string rozsah { get; set; } 
        public string typZk { get; set; } 
        public int? oborIdNum { get; set; }
        public List<Soubor> Files { get; set; } 
    }
}